osfoffline/gui/qt/preferences.py

```diff
@@ -148,13 +149,14 @@ def selector(self, selected_index):
 
             self._executor = QtCore.QThread()
             self.node_fetcher = NodeFetcher()
-            self.treeWidget.setCursor(QtCore.Qt.BusyCursor)
+            QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
```

Comment:
I would think so yes. For this reason I tried to write a decorator for the waiting cursor:

```python
def waiting_effects(function):
        def wrap(*args, **kwargs):
            QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
            try:
                function(*args, **kwargs)
            except Exception as e:
                raise e
                logger.exception(e.args[0])
            finally:
                QApplication.restoreOverrideCursor()
        return wrap
```

For some reason, however, none of the cursor changes happened with the above. Any other suggestions on how to avoid this?